apis/oam/v1alpha2/core_types.go

```diff
@@ -0,0 +1,375 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// A DefinitionReference refers to a CustomResourceDefinition by name.
+type DefinitionReference struct {
+	// Name of the referenced CustomResourceDefinition.
+	Name string `json:"name"`
+}
+
+// A WorkloadDefinitionSpec defines the desired state of a WorkloadDefinition.
+type WorkloadDefinitionSpec struct {
+	// Reference to the CustomResourceDefinition that defines this workload
+	// kind.
+	Reference DefinitionReference `json:"definitionRef"`
```

Comment:
I wonder if there is any reason to not keep the same name "definitionRef" as the json field name?

Reply:
It's Kubernetes convention for actual types and fields to use the full `Reference` name, rather than the abbreviated `Ref` that is used in the JSON fields.